connectors/sources/sharepoint_online.py

```diff
@@ -626,20 +626,82 @@ async def site_users(self, site_web_url):
         except NotFound:
             return
 
-    async def sites(self, parent_site_id, allowed_root_sites):
-        select = ""
+    async def sites(
+        self,
+        sharepoint_host,
+        allowed_root_sites,
+        enumerate_all_sites=True,
+        fetch_subsites=False,
+    ):
+        if allowed_root_sites == [WILDCARD] or enumerate_all_sites:
+            self._logger.debug(f"Looking up all sites to fetch: {allowed_root_sites}")
+            async for site in self._all_sites(sharepoint_host, allowed_root_sites):
+                yield site
+        else:
+            self._logger.debug(f"Looking up individual sites: {allowed_root_sites}")
```

Comment:
Probably you want to add a [get root site call](https://learn.microsoft.com/en-us/graph/api/site-get?view=graph-rest-1.0&tabs=http#get-the-tenants-root-site) in the else loop, assuming we always want to fetch root site.

Adding a new RCF `Fetch root site?` may be confusing. Imagine customers configured `*` in `site collections, and toggled off `Fetch root site?`, you need to exclude the root site in the above `if branch`.

Reply:
`*` is currently the only way to fetch a root site. Using a comma-separated list today would not fetch the root site either. I don't think we need to address that edge case in this PR, to avoid scope creep.

Reply:
Diana indicated in a [slack thread](https://elastic.slack.com/archives/C01795T48LQ/p1695390301379779?thread_ts=1695311571.172939&cid=C01795T48LQ) that this is a good candidate for a follow-up effort. I've filed: https://github.com/elastic/connectors-python/issues/1675